eth/tracers/printer.go

```diff
@@ -0,0 +1,103 @@
+package tracers
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+type Printer struct{}
+
+func NewPrinter() *Printer {
+	return &Printer{}
+}
+
+// CaptureStart implements the EVMLogger interface to initialize the tracing operation.
+func (p *Printer) CaptureStart(from common.Address, to common.Address, create bool, input []byte, gas uint64, value *big.Int) {
+	fmt.Printf("CaptureStart: from=%v, to=%v, create=%v, input=%v, gas=%v, value=%v\n", from, to, create, input, gas, value)
```

Comment:
All `[]byte` in the printer should be wrapped with `hexutil.Bytes(...)` to be printed nicely as hex.

```suggestion
func (p *Printer) CaptureStart(from common.Address, to common.Address, create bool, input []byte, gas uint64, value *big.Int) {
	fmt.Printf("CaptureStart: from=%v, to=%v, create=%v, input=%s, gas=%v, value=%v\n", from, to, create, hexutil.Bytes(input), gas, value)
```

I can provide a commit on top of your PR I have the change locally

Reply:
Actually it's not that better. Maybe we should even just shorten displayed bytes.

Reply:
I made the printer outputs a bit nicer. Sorry this is just a rudimentary example for me to see the hooks are actually working. I'm not planning to merge `Printer` and didn't pay so much attention to it. Please feel free to modify and push commits on top of this branch if it makes it easier for you to consume the logs.

Reply:
> I made the printer outputs a bit nicer. Sorry this is just a rudimentary example for me to see the hooks are actually working. I'm not planning to merge Printer and didn't pay so much attention to it. Please feel free to modify and push commits on top of this branch if it makes it easier for you to consume the logs.

Yeah my comment is more general, think it's quite ok having it the way you did for display purposes. I'll check to open something to at least render the `[]byte` as hex.

Reply:
Actually not needed, just saw you already pushed it :)